objcgen/processedtypes.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using IKVM.Reflection;
+using Type = IKVM.Reflection.Type;
+
+namespace ObjC {
+
+	// While processing user assemblies, we may come across conditions that will affect
+	// final code generation that we need to pass to the generation pass
+
+	public abstract class ProcessedBase {
+		public bool FallBackToTypeName { get; set; }
+	}
+
+	public class ProcessedMethod : ProcessedBase {
```

Comment:
We run into issues because type Method can't contain a property called method:
`
types.cs(21,21): Error CS0542: 'Method': member names cannot be the same as their enclosing type (CS0542) (objcgen)`